### ISSUE TITLE
Multiple References of Different Content Types

### ISSUE THREAD

**Issue by bretmorris** [davidpett/ember-data-contentful#24]
Hey @davidpett, first of all, thanks for building this! I am a pretty new Ember user so please forgive me if this is a really novice question.

How can one go about defining a model that has a field with many references of different content types?

In my case, I have a content type that has a field called "modules". The modules field is a "many reference" field type in Contentful. Specifically, I have set this field up to allow the user to select from one of several different content types.

At the moment, my model looks like this:

```
import Contentful from 'ember-data-contentful/models/contentful';
import attr from 'ember-data/attr';
import { belongsTo, hasMany } from 'ember-data/relationships';

export default Contentful.extend({
  title: attr('string'),
  subtitle: attr('string'),
  metadata: belongsTo('metadata'),
  modules: hasMany(['module-copy', 'model-testimonials'])
});
```

Note how there is a modules field that currently allows a user to reference many `module-copy`, and `module-testimonials` content types. How can I get this to work? Currently there are no errors, but the models have all `undefined` content for each field.

Many thanks!

**Comment by davidpett** [comment]
Hey @bretmorris, I have done something similar to this before, the idea is to have a base model (`module.js`) and then any module will then extend from that base module. I like to create a component for each of those modules as well to easily place them on the page with the `{{component}}` helper. Here is a gist that illustrates the point: https://gist.github.com/davidpett/b56e962e2109fa02671f17e238f1006b

**Comment by bretmorris** [comment]
@davidpett thanks so much for the gist! I almost have it working but for some reason the components are never printed to the page. Here's a gist I threw together. Anything stand out? https://gist.github.com/bretmorris/1c6787b45e5565260e98daaf8300b039

**Comment by bretmorris** [comment]
@davidpett here's what my ember dev tool data tab looks like: 

`modules`: https://cl.ly/251i0j3k2l0K
`module-copy`: https://cl.ly/0X3w150O2O39
`module-testimonial`: https://cl.ly/141a3l1V0V1E

The copy and testimonial modules look perfect but as you can see, the `modules` model shows everything as undefined. I believe this explains why the `component` helper isn't displaying the components on the page.

Any ideas?

**Comment by davidpett** [comment]
Looking at my app more (it's been a while) I have all of the attrs for all
sub modules in the module model. The sub modules are blank except for the
Module.extend. I can't remember why exactly that was, but it works

**Comment by bretmorris** [comment]
Yeah I realized that too @davidpett. Still not quite working for me. It looks like this line is the issue: 

```
moduleType: computed('contentType', function() {
    return get(this, 'contentType') ? `${get(this, 'contentType').dasherize()}` : null;
})
```

`null` is being returned every time because `get(this, 'contentType')` is always `undefined`.

**Comment by bretmorris** [comment]
Hey @davidpett wanted to point out that I created a stackoverflow for this http://stackoverflow.com/questions/43332852/ember-hasmany-with-multiple-models. This seems like it could be a very common issue someone would run into when using this plugin.

**Comment by davidpett** [comment]
Thanks @bretmorris, would it be possible to create a small ember twiddle or a private repo that you can share with me so I can see more of what's happening? I'm also frequently on the ember slack if you want to chat realtime

**Comment by bretmorris** [comment]
Absolutely. I just got on the ember slack as well and posted this question their too. I'll put together a twiddle tomorrow and send it your way. Really appreciate your help.

**Comment by bretmorris** [comment]
For anyone having a similar issue, downgrading to version `0.2.1` fixed the problem. @davidpett is aware and is looking into a fix sometime in the future.

**Comment by morganick** [comment]
@bretmorris Just so I can get a test case from a data stand point for this. You have a page content type with a many reference to multiple entries with various content types. I have a couple of hunches of what might be going on. I'll try to dive into this over the next couple days.


/cc @davidpett I broke it. I'll fix it 😅

**Comment by bretmorris** [comment]
exactly @morganick

**Comment by rogeraraujo90** [comment]
@bretmorris I know this issue is very old. But do you know if the current version has this fix? And how work with multiple content types in a same reference? I'm very lost with this.

@davidpett Is this repo still maintained?
